libcaf_test/caf/test/runnable.hpp

```diff
@@ -19,6 +19,18 @@
 
 namespace caf::test {
 
+namespace {
+
+template <class T0, class T1>
+inline void assert_integral_comparison(const T0& lhs, const T1& rhs) {
```

Comment:
Opening up the anonymous namespace in a header is kinda weird. Please move this to `runnable` as a (static) private member function and maybe rename to something a bit more "high level" like `assert_save_comparison`? Also, this doesn't need actual arguments since it's just operating on the types.